testsuite/ghc-config/ghc-config.hs

```diff
@@ -10,9 +10,9 @@ main = do
   let fields = read info :: [(String,String)]
   getGhcFieldOrFail fields "HostOS" "Host OS"
   getGhcFieldOrFail fields "WORDSIZE" "Word size"
-  getGhcFieldOrFail fields "TARGETPLATFORM" "Target platform"
-  getGhcFieldOrFail fields "TargetOS_CPP" "Target OS"
-  getGhcFieldOrFail fields "TargetARCH_CPP" "Target architecture"
+  getGhcFieldOrFail fields "TARGETPLATFORM" "Host platform"
+  getGhcFieldOrFail fields "TargetOS_CPP" "Host OS"
+  getGhcFieldOrFail fields "TargetARCH_CPP" "Host architecture"
   getGhcFieldOrFail fields "RTSWay" "RTS way"
 
```

Comment:
Is us filling in the Target platform/OS/architecture wrong?

Reply:
If we get the information from the RTS, we shouldn't use "target" info as it doesn't make sense for the RTS to have those. If we get the information from the compiler, then it would make sense to use target info.

See also https://gitlab.haskell.org/ghc/ghc/-/issues/24058#note_529995

Reply:
I'll fix it upstream too

Reply:
https://gitlab.haskell.org/ghc/ghc/-/merge_requests/13967

Reply:
I guess we'll need to backport that sufficiently far back as well 😬